onnxruntime/python/tools/microbench/benchmark.py

```diff
@@ -1,76 +1,111 @@
+from abc import ABC, abstractmethod
 from argparse import ArgumentParser
 import time
 import numpy
 import onnxruntime as ort
 import torch
 
 
-def add_arguments(parser: ArgumentParser):
-    parser.add_argument("--provider", required=False, type=str, default="rocm", help="Execution provider to use")
-    parser.add_argument("--precision", required=False, type=str, default="fp16", help="Number format to use")
-    parser.add_argument('--profiling', type=bool, default=False, help='If enable profiling')
-
-
-def create_input_output_tensors(inputs, outputs):
-    device = "cuda"
-    input_tensors = {name: torch.from_numpy(array).to(device) for name, array in inputs.items()}
-    output_tensors = {name: torch.from_numpy(array).to(device) for name, array in outputs.items()}
-    return input_tensors, output_tensors
-
-
 def numpy_type(torch_type):
     type_map = {torch.float32: numpy.float32,
                 torch.float16: numpy.float16}
     return type_map[torch_type]
 
 
-def create_io_binding(sess, input_tensors, output_tensors):
-    io_binding = sess.io_binding()
-
-    for name, tensor in input_tensors.items():
-        io_binding.bind_input(name, tensor.device.type, 0, numpy_type(tensor.dtype), tensor.shape, tensor.data_ptr())
-
-    for name, tensor in output_tensors.items():
-        io_binding.bind_output(name, tensor.device.type, 0, numpy_type(tensor.dtype), tensor.shape, tensor.data_ptr())
-  
-    return io_binding
-
-
-def create_session(onnx_file, args):
-    sess_opt = ort.SessionOptions()
-    sess_opt.enable_profiling = args.profiling
-    if args.provider == "rocm":
-        execution_provider = ["ROCMExecutionProvider"] 
-    elif args.provider == "cuda":
-        execution_provider = ["CUDAExecutionProvider"] 
-    else:
-        raise ValueError(f"The script doesn't support provider type '{args.provider}' yet.")
-
-    sess = ort.InferenceSession(onnx_file, sess_options=sess_opt, providers=execution_provider)
-
-    if args.provider == "rocm":
-        assert 'ROCMExecutionProvider' in sess.get_providers()
-    elif args.provider == "cuda":
-        assert 'CUDAExecutionProvider' in sess.get_providers()
-
-    return sess
- 
+def add_arguments(parser: ArgumentParser):
+    parser.add_argument("--provider", required=False, type=str, default="rocm", help="Execution provider to use")
```

Comment:
>rocm [](http://example.com/codeflow?start=73&length=4)

on v100, if no args specified, we'll get unsupported EP (i.e. default is rocm). Perhaps we can add a small message on usage, or -h to print what's the current available params.
NIT - perhaps we can add it next time when you revisit this again.

Reply:
Thanks; will add argument choices in the next PR.